src/components/fields/MarkerColor.js

```diff
@@ -0,0 +1,142 @@
+import Field from './Field';
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import {connectToContainer} from 'lib';
+import RadioBlocks from '../widgets/RadioBlocks';
+import Color from './Color';
+import DataSelector from './DataSelector';
+import Colorscale from './Colorscale';
+import {MULTI_VALUED, COLORS} from 'lib/constants';
+
+class UnconnectedMarkerColor extends Component {
+  constructor(props, context) {
+    super(props, context);
+
+    let type = null;
+    if (
+      !props.container.marker ||
+      (props.container.marker && !props.container.marker.colorsrc)
+    ) {
+      type = 'constant';
+    } else if (
+      props.container.marker &&
+      Array.isArray(props.container.marker.color) &&
+      props.fullContainer.marker &&
+      Array.isArray(props.fullContainer.marker.color)
```

Comment:
that's a very safe way to check :)
I guess that it's because we really want to check that it's user set, that's why we're looking in props.container.

And just to make sure, we're checking in fullContainer too, to see if the user hasn't given some crazy inputs that were not accepted by plotly.js.

ok!

Reply:
You're laughing but multivalue mechanism makes these things complicated. If you have one trace with variable color and one with constant, `container` could be either array or `MULTI_VALUE`, while `fullContainer` will be array for sure. When both traces are variable, both will be arrays.